pkg/trace/stats/aggregation.go

```diff
@@ -80,15 +88,54 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
 			Synthetics: synthetics,
 		},
 	}
-	if enablePeerSvcAgg {
-		agg.PeerService = s.Meta[tagPeerService]
+	var peerTags []string
+	if spanKindIsClientOrProducer(agg.SpanKind) {
+		if enablePeerSvcAgg {
+			agg.PeerService = s.Meta[tagPeerService]
+		}
+		peerTags = getMatchingPeerTags(s, peerTagKeys)
+		agg.PeerTagsHash = peerTagsHash(peerTags)
 	}
-	return agg
+	return agg, peerTags
+}
+
+func getMatchingPeerTags(s *pb.Span, peerTagKeys []string) []string {
+	if len(peerTagKeys) == 0 {
+		return nil
+	}
+	var pt []string
+	for _, t := range peerTagKeys {
+		v, ok := s.Meta[t]
+		if ok {
+			pt = append(pt, t+":"+v)
+		}
+	}
+	if len(pt) == 0 {
+		return nil
+	}
```

Comment:
No need for this. If the loop never matches anything, then `pt` will be nil anyway